test/e2e/clusterclass_changes.go

```diff
@@ -426,6 +427,131 @@ func modifyMachineDeploymentViaClusterClassAndWait(ctx context.Context, input mo
 	}
 }
 
+// modifyMachinePoolViaClusterClassAndWaitInput is the input type for modifyMachinePoolViaClusterClassAndWait.
+type modifyMachinePoolViaClusterClassAndWaitInput struct {
+	ClusterProxy                                  framework.ClusterProxy
+	ClusterClass                                  *clusterv1.ClusterClass
+	Cluster                                       *clusterv1.Cluster
+	ModifyBootstrapConfigTemplateFields           map[string]interface{}
+	ModifyInfrastructureMachinePoolTemplateFields map[string]interface{}
+	WaitForMachinePools                           []interface{}
+}
+
+// modifyMachinePoolViaClusterClassAndWait modifies the BootstrapConfigTemplate of MachinePoolClasses of a ClusterClass
+// by setting ModifyBootstrapConfigTemplateFields and waits until the changes are rolled out to the MachinePools of the Cluster.
+// NOTE: This helper is really specific to this test, so we are keeping this private vs. adding it to the framework.
+func modifyMachinePoolViaClusterClassAndWait(ctx context.Context, input modifyMachinePoolViaClusterClassAndWaitInput) {
```

Comment:
Just for a follow-up PR independent of the upcoming 1.6 release.

I think we also should call this function in this test (~ l. 183)

Reply:
We should also add corresponding "Modify" fields to ClusterClassChangesSpecInput and set them in clusterclass_changes_test.go